### DISCUSSION TITLE
How to draw an array containing a lot of rectangles of a specific size？

### DISCUSSION THREAD

**Discussion by YuleiDing** [gdsfactory/gdsfactory#2324] [Q&A] [answered]
I want to use the generic PDK of component.array to draw an array containing a lot of rectangles of a specific size. For example, the size of the rectangle is 5*3
This is my code:
"R1 = gf.components.array(component='rectangle', spacing=[150.0, 150.0], columns=6, rows=10, add_ports=True) "
I don't know how to change the size of the 'rectangle'，does anyone know how to do it ？

**Comment by joamatab** [comment, marked as answer]
This is the most elegant way

```python
import gdsfactory as gf
from functools import partial 

R15 = partial(gf.components.rectangle, size=(15, 15))
R1 = gf.components.array(component=R15, spacing=(150.0, 150.0), columns=6, rows=10, add_ports=True)
```

This also works and is simpler

```python
import gdsfactory as gf

R15 = gf.components.rectangle(size=(15, 15))
R1 = gf.components.array(component=R15, spacing=(150.0, 150.0), columns=6, rows=10, add_ports=True)
R1.show()
```

**Reply by YuleiDing** [reply]
Thank you！
